src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -98,7 +98,7 @@ class RustProjectSettingsPanel(
                 val toolchain = RustToolchain(Paths.get(pathToToolchain))
                 val rustcVersion = toolchain.queryVersions().rustc?.semver
                 val rustup = toolchain.rustup
-                val stdlibLocation = rustup?.getStdlibFromSysroot()?.presentableUrl
+                val stdlibLocation = toolchain.getStdlibFromSysroot(cargoProjectDir)?.presentableUrl
```

Comment:
Looks like it's not enough to just fetch stdlib path here
You also need to set this path to `pathToStdlibField`. Now we set the corresponding path only when `rustup != null`

Reply:
Hm, yeah, looks like `if (rustup == null)` logic is pretty ingrained everywhere :-( `CargoProjectImpl.refreshStdlib` does this as well. 

I **might** try to fix this properly if I have time to spare, but this is pretty unlikely: it seems like we might want to do a medium-sized refactoring of this logic, and the relevant info is out of cache for me :)